esprima.js

```diff
@@ -4023,18 +4023,20 @@
         if (match('=')) {
             lex();
             init = isolateCoverGrammar(parseAssignmentExpression);
-        } else if (id.type !== Syntax.Identifier) {
-            expect('=');
+        } else {
+            if (!(options && options.inFor) && (id.type !== Syntax.Identifier)) {
+                expect('=');
+            }
         }
 
         return node.finishVariableDeclarator(id, init);
     }
 
-    function parseVariableDeclarationList() {
+    function parseVariableDeclarationList(options) {
         var list = [];
 
         do {
-            list.push(parseVariableDeclaration());
+            list.push(parseVariableDeclaration(options));
             if (!match(',')) {
```

Comment:
I think at this place, there is no sufficient information yet that it will be a for-in.

The check is carried out a bit later:
```js
                if (declarations.length === 1 && declarations[0].init === null && matchKeyword('in')) {
```